eo-maven-plugin/src/test/java/org/eolang/maven/RegisterMojoIT.java

```diff
@@ -0,0 +1,247 @@
+                    ).getBytes()
+                );
+                f.exec("eo:register");
+                MatcherAssert.assertThat(
+                    "Old resolved files must were removed, but it didn't",
+                    temp.resolve(
+                        String.format(
+                            "target/eo/%s",
+                            ResolveMojo.DIR
+                        )
+                    ).toFile().exists(),
+                    Matchers.is(false)
+                );
+            }
+        );
+    }
+
+    @Test
+    void removesOldForeignFile(@Mktmp final Path temp) throws Exception {
+        new Farea(temp).together(
+            f -> {
+                f.clean();
+                f.files().file("src/main/eo/foo.eo").write(
+                    String.join(
+                        "\n",
+                        "+package org.eolang",
+                        "",
+                        "# In this program, we refer to the 'String'",
+                        "[] > foo",
+                        "  \"42\" > @"
+                    ).getBytes()
+                );
+                RegisterMojoIT.configureFarea(f);
+                f.exec("eo:register", "eo:parse", "eo:shake", "eo:probe", "eo:pull");
+                f.files().file("src/main/eo/foo.eo").write(
+                    String.join(
+                        "\n",
+                        "+package org.eolang",
+                        "",
+                        "# In this program, we refer to the 'Number'.",
+                        "[] > foo",
+                        "  42 > @"
+                    ).getBytes()
+                );
+                f.exec("eo:register", "eo:parse", "eo:shake", "eo:probe", "eo:pull");
+                final TjSmart foreign = RegisterMojoIT.foreign(
+                    temp.resolve("target/eo-foreign.csv")
+                );
+                MatcherAssert.assertThat(
+                    "Foreign must contain only 3 references to objects, but it doesn't",
+                    foreign.size(),
+                    Matchers.equalTo(3)
+                );
+                MatcherAssert.assertThat(
+                    "Foreign must contain reference to the Number object, but it doesn't",
+                    foreign.getById("org.eolang.number").exists("id"),
+                    Matchers.is(true)
+                );
+                MatcherAssert.assertThat(
+                    "Foreign must contain reference to the Bytes object, but it doesn't",
+                    foreign.getById("org.eolang.bytes").exists("id"),
+                    Matchers.is(true)
+                );
+                MatcherAssert.assertThat(
+                    "Foreign must contain reference to the current object, but it doesn't",
+                    foreign.getById("foo").exists("id"),
+                    Matchers.is(true)
+                );
+                Assertions.assertThrows(
+                    NoSuchElementException.class,
```

Comment:
@Suban05 checking the presence of an element by exception is not a really good practice, let's use the method `.select()` with predicate and the the size of result list